platzky/shortcodes/shortcode.py

```
@@ -0,0 +1,182 @@
+    name: str
+    description: str
+    attributes: ClassVar[ShortcodeAttrs] = ShortcodeAttrs([])
+    example: str = ""
+
+    def __init_subclass__(cls, **kwargs: object) -> None:
+        super().__init_subclass__(**kwargs)
+        if "name" in cls.__dict__ and not _VALID_SHORTCODE_NAME_RE.match(cls.name):
+            raise ValueError(
+                f"Shortcode name {cls.name!r} is invalid. "
+                "Use only letters, digits, hyphens, and underscores, starting with a letter."
+            )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)